pkg/serverconfig/metrics.go

```diff
@@ -1,12 +1,51 @@
 package serverconfig
 
 import (
+	"context"
+	"net/http"
+	"time"
+
 	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
 )
 
+// We don't expect that the plugin metrics changes regullary (without a new console rollout).
+const updateConsolePluginInterval = 6 * time.Hour
+
+var consolePluginResource = schema.GroupVersionResource{
+	Group:    "console.openshift.io",
+	Version:  "v1",
+	Resource: "consoleplugins",
+}
+
 type Metrics struct {
-	pluginsInfo      *prometheus.GaugeVec
+	config           *Config
 	perspectivesInfo *prometheus.GaugeVec
+	pluginsInfo      *prometheus.GaugeVec
+	// Keep the last info so that it is possible to report zero for removed ConsolePlugins.
+	lastPluginInfo map[PluginVendor]map[PluginState]int
+}
+
+// Reduce cardinality by grouping all perspectives by a 'group name' or a vendor.
+type PerspectiveGroupOrVendor string
+
+const (
+	// Seperate admin, dev and acm.
+	PerspectiveGroupAdmin PerspectiveGroupOrVendor = "admin"
+	PerspectiveGroupDev   PerspectiveGroupOrVendor = "dev"
+	PerspectiveGroupACM   PerspectiveGroupOrVendor = "acm"
```

Comment:
Not sure about having ACM here, since the priorities have shifted.
@spadgett thoughts ?